InventoryManagementSystem_SQL/DataBaseOperations/DBOperations.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System.Data.SqlClient;
+
+namespace InventoryManagementSystem
+{
+    public class DBOperations : SqlServerDatabaseInitializer, IProduct
+    {
+        public async Task AddProductAsync(Product newProduct)
+        {
+            var query = $"""
+                INSERT INTO Products
+                VALUES (@{nameof(newProduct.Name)}, @{nameof(newProduct.Price)}, 
+                @{nameof(newProduct.Quantity)})
+                """;
+
+            using (var connection = new SqlConnection(_connectionString))
+            {
+                await connection.OpenAsync();
+
+                using var sqlCommand = new SqlCommand(query, connection);
+                sqlCommand.Parameters.AddWithValue($"@{nameof(newProduct.Name)}", newProduct.Name);
+                sqlCommand.Parameters.AddWithValue($"@{nameof(newProduct.Price)}", newProduct.Price);
+                sqlCommand.Parameters.AddWithValue($"@{nameof(newProduct.Quantity)}", newProduct.Quantity);
+                await sqlCommand.ExecuteNonQueryAsync();
+            }
+        }
+
+        public async Task EditProductNameAsync(string oldName, string newName)
+        {
+            var query = $"""
+                UPDATE Products
+                SET name = @{nameof(newName)}
+                WHERE name = @{nameof(oldName)}
+                """;
+
+            using (var connection = new SqlConnection(_connectionString))
+            {
+                await connection.OpenAsync();
+
+                using var sqlCommand = new SqlCommand(query, connection);
+                sqlCommand.Parameters.AddWithValue($"@{nameof(newName)}", newName);
+                sqlCommand.Parameters.AddWithValue($"@{nameof(oldName)}", oldName);
+                await sqlCommand.ExecuteNonQueryAsync();
+            }
+        }
+
+        public async Task EditProductPriceAsync(string productName, decimal newPrice)
+        {
+            var query = $"""
+                UPDATE Products
+                SET price = @{nameof(newPrice)}
+                WHERE name = @{nameof(productName)}
+                """;
+
+            using (var connection = new SqlConnection(_connectionString))
+            {
+                await connection.OpenAsync();
+
+                using var sqlCommand = new SqlCommand(query, connection);
+                sqlCommand.Parameters.AddWithValue($"@{nameof(newPrice)}", newPrice);
+                sqlCommand.Parameters.AddWithValue($"@{nameof(productName)}", productName);
+                await sqlCommand.ExecuteNonQueryAsync();
+            }
+        }
+
+        public async Task EditProductQuantityAsync(string productName, int newQuantity)
+        {
+            var query = $"""
+                UPDATE Products
+                SET quantity = @{nameof(newQuantity)}
+                WHERE name = @{nameof(productName)}
+                """;
```

Comment:
I think it would be good to have separate folder for each Model with 2 files each one for queries as get and one for commands as update
have method for each returning the query 
as ` var query = ProductQueries.UpdateProductsQuantityByName(nameof(productName), nameof(newQuantity))`